python/cudf/cudf/core/column/categorical.py

```diff
@@ -1417,8 +1417,23 @@ def _get_decategorized_column(self) -> ColumnBase:
 
     def copy(self, deep: bool = True) -> CategoricalColumn:
         if deep:
+
             copied_col = libcudf.copying.copy_column(self)
-            copied_cat = libcudf.copying.copy_column(self.dtype._categories)
+            if not is_interval_dtype(self.dtype._categories):
+                copied_cat = libcudf.copying.copy_column(
+                    self.dtype._categories
+                )
+            else:
+                copied_struct = libcudf.copying.copy_column(
+                    self.dtype._categories
+                )
+                # avoiding circular import since IntervalColumn
+                # inherits from StructColumn
+                from cudf.core.column.interval import IntervalColumn
+
+                copied_cat = IntervalColumn.from_struct_column(
+                    copied_struct, closed=self.dtype._categories.closed
+                )
```

Comment:
Same logic but shorter code?
```suggestion
            copied_cat = libcudf.copying.copy_column(
                self.dtype._categories
            )
            if not is_interval_dtype(self.dtype._categories):
                # avoiding circular import since IntervalColumn
                # inherits from StructColumn
                from cudf.core.column.interval import IntervalColumn

                copied_cat = IntervalColumn.from_struct_column(
                    copied_cat, closed=self.dtype._categories.closed
                )
```